mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -171,11 +171,22 @@ public void rollbackDropTable(org.apache.hadoop.hive.metastore.api.Table hmsTabl
   @Override
   public void commitDropTable(org.apache.hadoop.hive.metastore.api.Table hmsTable, boolean deleteData) {
     if (deleteData && deleteIcebergTable) {
-      if (!Catalogs.hiveCatalog(conf)) {
-        LOG.info("Dropping with purge all the data for table {}.{}", hmsTable.getDbName(), hmsTable.getTableName());
-        Catalogs.dropTable(conf, catalogProperties);
-      } else {
-        CatalogUtil.dropTableData(deleteIo, deleteMetadata);
+      try {
+        if (!Catalogs.hiveCatalog(conf)) {
+          LOG.info("Dropping with purge all the data for table {}.{}", hmsTable.getDbName(), hmsTable.getTableName());
+          Catalogs.dropTable(conf, catalogProperties);
+        } else {
+          // if metadata folder has been deleted already (Hive 4 behaviour for purge=TRUE), simply return
+          if (!deleteIo.newInputFile(deleteMetadata.location()).exists()) {
```

Comment:
nit: Why not:
```
if (deleteIo.newInputFile(deleteMetadata.location()).exists()) {
  CatalogUtil.dropTableData(deleteIo, deleteMetadata);
}
```

This way we can avoid returning in the middle of the code which is often hard to read, also maybe it is easier to understand.

Reply:
Yes, that's cleaner :)